symengine/lib/symengine_wrapper.pyx

```diff
@@ -1740,6 +1740,13 @@ cdef class DenseMatrix(MatrixBase):
         def __get__(self):
             return self.nrows()*self.ncols()
 
+    def reshape(self, rows, cols):
+        if len(self) != rows*cols:
+            raise ValueError("Invalid reshape parameters %d %d" % (rows, cols))
+        cdef symengine.vec_basic v
+        matrix_to_vec(self, v)
+        self.thisptr = new symengine.DenseMatrix(rows, cols, v)
```

Comment:
There is a reshape method in C++, you can use that. Also, in sympy, this method returns a new matrix instead of changing the current one.